db/changes_test.go

```diff
@@ -22,6 +22,87 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// Unit test for CBG-1326
+func TestLogIfChannelsDroppedFromList(t *testing.T) {
+	db := setupTestDB(t)
+	defer db.Close()
+
+	defer base.SetUpTestLogging(base.LevelWarn, base.KeyAll)()
+
+	auth := db.Authenticator()
+	user, _ := auth.NewUser("test", "pass", channels.SetOf(t, "ch1", "ch3"))
+	require.NoError(t, auth.Save(user))
+
+	// +1 warning when retrieving changes from channels where the user has no access to some of them channels
+	t.Run("Warn when channels dropped from list", func(t *testing.T) {
+		changes, err := db.GetChanges(base.SetOf("*"), getZeroSequence())
+		lastSeq := getLastSeq(changes)
+		lastSeq, _ = db.ParseSequenceID(lastSeq.String())
+
+		_, _, err = db.Put("doc1", Body{"channels": []string{"ch1"}})
+		require.NoError(t, err)
+		_, _, err = db.Put("doc2", Body{"channels": []string{"ch2"}})
+		require.NoError(t, err)
+		_, _, err = db.Put("doc3", Body{"channels": []string{"ch3"}})
+		require.NoError(t, err)
+
+		db.WaitForPendingChanges(context.Background())
+
+		db.user, _ = auth.GetUser("test")
+
+		warnCount := base.SyncGatewayStats.GlobalStats.ResourceUtilizationStats().WarnCount.Value()
+
+		ch, err := db.GetChanges(base.SetOf("ch1", "ch2", "ch3"), ChangesOptions{Since: lastSeq})
```

Comment:
Any reason we're doing Since sequences like this?
I think it'd be simpler if the setup was done in each of the test runs (perhaps in a function if they're the same) and then we can just do since 0. Avoids doing the initial GetChanges to get lastSeq and all that. Would save you a few lines.